### PR TITLE
[ Tezos ] Fix undelegation crash

### PR DESCRIPTION
**Pull request by leoneparise**
## Description
Fix undelegation crash for tezos and add more test cases

### CODE REVIEW

File: src/Tezos/Forging.cpp [hunk 1]
```diff
@@ -84,14 +84,14 @@ Data forgeOperation(const Operation& operation) {
         auto publicKey = PublicKey(operation.reveal_operation_data().public_key(), TWPublicKeyTypeED25519);
         auto forgedPublicKey = forgePublicKey(publicKey);
 
-        /* Uncomment for debugging */
+        /* Uncomment for debugging
```
Comment (hewigovens) [inline src/Tezos/Forging.cpp]: Can we remove comments for debugging? It’s not document either

File: src/Tezos/Forging.cpp [hunk 2]
```diff
@@ -123,8 +121,11 @@ Data forgeOperation(const Operation& operation) {
         append(forged, forgedGasLimit);
         append(forged, forgedStorageLimit);
         if (!delegate.empty()) {
+            auto forgedPublicKeyHash = forgePublicKeyHash(delegate);
+            // auto forgedPublicKeyHashHex = hex(forgedPublickKeyHash.begin(), forgedPublickKeyHash.end()); // Uncomment for debugging
```
Comment (hewigovens) [inline src/Tezos/Forging.cpp]: Also here

File: src/Tezos/Forging.cpp (hunk 1 above)
Comment (leoneparise) [reply src/Tezos/Forging.cpp]: It helped me a lot when I was having issues to forge those binaries.

Comment (hewigovens) [reply src/Tezos/Forging.cpp]: yes, I'm used to have them in my git stash, git stash pop if you want to debug them

**Pull request merged**